### DISCUSSION TITLE
Extend CLI's `generate entity` command with an option to not change `mod.rs` file

### DISCUSSION THREAD

**Discussion by teenjuna** [SeaQL/sea-orm#1196] [Ideas]
## Motivation

Codegen has some problems:
1. If I'm running the generate entity without `-t`, codegen overwrites everything, including custom code I added to entities (for example, implementation for `ActiveModelBehaviour` or `Linked`).
2. If I'm running the generate entity with `-t`, it overwrites `mod.rs` file, including only the entities for specified tables.

In both cases, one must use copy-paste and often seek for git to recover code.

## Proposed Solutions

One relatively easy solution that fixes (2) is to add an option for CLI that tells the codegen to not touch the `mod.rs` file. Instead, print a message asking the user to not forget to include the entity module if it's not already there. The main question is: how to name this option? `--only-entity-files` seems to be to verbose.

Another solution is to make the behaviour described above default when `-t` is present (and maybe add an option to modify `mod.rs`, something like `--with-mod-rs`).

Thoughts?

**Comment by teenjuna** [comment]
Hey @billy1624, this issue suggests a couple of ways to implement this. Could you tell me which is more suitable in your opinion, so I can start working on this?

**Comment by teenjuna** [comment]
I guess that `--only-entity-files` is the way to go since not only `mod.rs` file can be changed, but also `prelude.rs` and `sea_orm_active_enum.rs`

**Comment by tsar-boomba** [comment]
I worked around this by outputting entities to `entity/src/entities` using the `-o` option instead of `entity/src`, leaving anything outside of `entity/src/entities` untouched.

**Comment by billy1624** [comment]
Hey @teenjuna, sorry for the delay.

I think @tsar-boomba, made a good point. Outputting the entities in another directory then pick the bits back to the original entity directory.

However, a better solution would be track the project with Git and outputting the entity files directly on the corresponding directory. Then, check the Git diff and commit the useful bits.

**Comment by billy1624** [comment]
> I guess that `--only-entity-files` is the way to go since not only `mod.rs` file can be changed, but also `prelude.rs` and `sea_orm_active_enum.rs`

I think having a flag to output entity file only isn't ideal. Because it might need to output `sea_orm_active_enum.rs` if the target entity contains Enum. Also, it might overridden the `sea_orm_active_enum.rs` and removing other Enums that is used by other entities but not the target entity.
